react-components/intro-to-react-js.md

```diff
@@ -0,0 +1,101 @@
+# Intro to React JS
+
+## Learning Goals
+
+- What is React?
+- Why do we learn React?
+- Who is using React?
+- Intro to the Virtual DOM
+- Intro to Components
+
+## Introduction
+
+Consider a social media website, such as [Reddit.com](https://www.reddit.com/r/reactjs/). Every given web page needs to do soooo many things! To name a few, each web page needs to:
+
+- List posts, the number of upvotes, and the comments
+- Display upvote and downvote buttons
+- Handle upvote and downvote buttons
+- List related communities
+- Handle the buttons to "join" those communities
+- Display and handle the UI to create a new post
+
+![](../assets/react-components_intro-to-react-js_reddit.png)
+
+Imagine creating all of this functionality using HTML and vanilla JavaScript. It wouldn't be impossible, but our resulting code is probably:
+
+- Huge! There's a lot of code in that logic to write!
+- Unorganized. Is there one file that holds all the JS? Are there multiple files? How are they organized?
+- Full of repetition. We need to create event handlers all the time. Is there a way to simplify this repetition?
+
+Great software is maintainable, functional, and performant. How can we strive towards those values in our front-end code?
+
+## Vocabulary and Synonyms
+
+| Vocab               | Definition                                                                                                                                                                                               | How to Use in a Sentence                                                                                                                                                                                               |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| User Interface (UI) | The elements that a user uses in order to interact with a machine. UI elements are usually visual, but UI also describes _how_ a user interacts with a webapp, such as buttons, text fields, or sliders. | "This UI was easy to use and enjoyable, because the text was readable, and the buttons were intuitive," "This UI was difficult to use, because I didn't understand the navigation menu, and the data was unorganized." |
```

Comment:
rephrase to avoid user uses
```suggestion
| User Interface (UI) | The elements with which a user interacts in order to make a machine do something. UI elements are usually visual, but UI also describes _how_ a user interacts with a webapp, such as buttons, text fields, or sliders. | "This UI was easy to use and enjoyable, because the text was readable, and the buttons were intuitive," "This UI was difficult to use, because I didn't understand the navigation menu, and the data was unorganized." |
```